src/compiler/checker.ts

```diff
@@ -15712,6 +15712,21 @@ namespace ts {
             if (getModifierFlags(node) & ModifierFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            // Is this the correct time to make assertions against the inheritance chain?

```

Comment:
Around line 18164 (look for `checkKindsOfPropertyMemberOverrides(type, baseType);`) / 18261 might be better; here the base type members are already in scope and we are iterating through the class properties.